tests/snapshot_test.c

```diff
@@ -95,6 +107,12 @@ void setup_process_state()
    CHECK_SYSCALL(epollfd = epoll_create(1));
    // leave a gap in FS space for recovery
    CHECK_SYSCALL(close(tmpfd));
+
+   struct sigaction act = {
+       .sa_sigaction = test_sigaction,
+       .sa_flags = SA_SIGINFO,
+   };
+   CHECK_SYSCALL(sigaction(SIGUSR1, &act, NULL));
    return;
```

Comment:
Could you add some code here to pull the sigaction structure back from the kernel and verify that SA_RESTORER has been set by the c library?

Reply:
done